src/java/org/apache/cassandra/db/guardrails/Guardrails.java

```diff
@@ -51,11 +51,11 @@ public final class Guardrails implements GuardrailsMBean
     /**
      * Guardrail on the total number of user keyspaces.
      */
-    public static final Threshold keyspaces =
-    new Threshold("keyspaces",
+    public static final MaxThreshold keyspaces =
```

Comment:
If we add `Threshold#guard` and `Threshold#triggersOn` abstract methods we could keep the type of these instances as `Threshold`, with a `MaxThreshold` or `MinThreshold` constructor, since callers don't strictly need to know if the type of threshold. No particular preference on this, though.

Reply:
I like types as its clear at usage, but agree that 90% of the code should be in a shared base type of Threshold